### PR TITLE
Fix device mismatch bug for categorical accuracy metric in distributed training

### PR DESCRIPTION
**Pull request by wlhgtc**
I got same error( `RuntimeError: Tensors must be CUDA and dense` )  in #4623 when I used distributed training with **allennlp 1.1.0** and **pytorch 1.6**.
I read your PR in #4570 , seem you only fix it in `Average` Class. When I use `CategoricalAccuracy `, the same error occur. 
So I try to fix it by move these tensors to GPU (same in `Average`), it works well.
Maybe you could help me check my solution, then I could update all related Metrics Classes ?

### CODE REVIEW

**Comment by wlhgtc** [comment]
@dirkgr  @matt-gardner

File: CHANGELOG.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Removed `common.util.is_master` function.
 
 ### Fixed
-
+- Fix distributed metrics error.
```
Comment (epwalsh) [inline CHANGELOG.md:66]:
```suggestion

- Fix CUDA/CPU device mismatch bug during distributed training for categorical accuracy metric.
```

**Review by epwalsh — CHANGES_REQUESTED** [review]
Thanks for catching this @wlhgtc!

File: CHANGELOG.md (hunk 1 above)
Comment (wlhgtc) [reply CHANGELOG.md:66]: After re-check other metrics, seem the operation is missed only in `CategoricalAccuracy `.
But this operation is done by different method in different classes.
Do you think we need a universal method for all tensors ?

**Review by epwalsh — APPROVED** [review]

**Pull request merged**